src/custom/state/orders/updaters/PendingOrdersUpdater.ts

```diff
@@ -35,6 +42,134 @@ function _getNewlyPreSignedOrders(allPendingOrders: Order[], signedOrdersIds: Or
+
+      // Get safe info
+      safeTransactionPromise
+        .then(async (safeTransaction) => {
+          console.log('[PendingOrdersUpdater] Update Gnosis Safe transaction info: ', safeTransaction)
+          updatePresignGnosisSafeTx({ orderId: order.id, chainId, safeTransaction })
+        })
+        .catch((error) => {
+          if (!error.isCancelledError) {
+            console.error(
+              `[PendingOrdersUpdater] Failed to check Gnosis Safe tx hash: ${presignGnosisSafeTxHash}`,
+              error
+            )
+          }
+        })
+    })
+
+  await Promise.all(getSafeTxPromises)
+}
+
+interface UpdateOrdersParams {
+  chainId: ChainId
+  orders: Order[]
+
+  // Actions
+  fulfillOrdersBatch: FulfillOrdersBatchCallback
+  expireOrdersBatch: ExpireOrdersBatchCallback
+  cancelOrdersBatch: CancelOrdersBatchCallback
+  presignOrders: PresignOrdersCallback
+  updatePresignGnosisSafeTx: UpdatePresignGnosisSafeTxCallback
+  getSafeInfo: GetSafeInfo
+}
+
+async function _updateOrders({
+  chainId,
+  orders,
+
+  // Actions
+  fulfillOrdersBatch,
+  expireOrdersBatch,
+  cancelOrdersBatch,
+  presignOrders,
+  updatePresignGnosisSafeTx,
+  getSafeInfo,
+}: UpdateOrdersParams): Promise<void> {
+  // Exit early when there are no pending orders
+  if (orders.length === 0) {
+    return
+  }
+
+  // Iterate over pending orders fetching API data
+  const unfilteredOrdersData = await Promise.all(
+    orders.map(async (orderFromStore) => fetchOrderPopupData(orderFromStore, chainId))
+  )
+
+  // Group resolved promises by status
+  // Only pick the status that are final
+  const { fulfilled, expired, cancelled, presigned } = unfilteredOrdersData.reduce<
+    Record<OrderTransitionStatus, OrderLogPopupMixData[]>
+  >(
+    (acc, { status, popupData }) => {
+      popupData && acc[status].push(popupData)
+      return acc
+    },
+    { fulfilled: [], expired: [], cancelled: [], unknown: [], presigned: [], pending: [] }
+  )
+
+  if (presigned.length > 0) {
+    // Only mark as presigned the orders we were not aware of their new state
+    const presignedOrderIds = presigned as OrderID[]
+    const ordersPresignaturePendingSigned = _getNewlyPreSignedOrders(orders, presignedOrderIds)
+
+    if (ordersPresignaturePendingSigned.length > 0) {
+      presignOrders({
+        ids: ordersPresignaturePendingSigned,
+        chainId,
+      })
+    }
+  }
+
+  if (expired.length > 0) {
+    expireOrdersBatch({
+      ids: expired as OrderID[],
+      chainId,
+    })
+  }
+
+  if (cancelled.length > 0) {
+    cancelOrdersBatch({
+      ids: cancelled as OrderID[],
+      chainId,
+    })
+  }
+
+  if (fulfilled.length > 0) {
+    fulfillOrdersBatch({
+      ordersData: fulfilled as OrderFulfillmentData[],
+      chainId,
+    })
+  }
+
+  // Update the presign Gnosis Safe Tx info (if applies)}
+  await _updatePresignGnosisSafeTx(chainId, orders, getSafeInfo, updatePresignGnosisSafeTx)
```

Comment:
This step is independent of all the rest of the API update stuff, no?
Maybe it should be executed in another flow/function/cycle?